extensions/css-language-features/client/src/node/dropIntoEditor.ts

```diff
@@ -0,0 +1,89 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { Mimes, extractUriList } from './shared';
+
+class DropIntoEditorProvider implements vscode.DocumentDropEditProvider {
+	readonly id = 'relativePath';
+	readonly dropMimeTypes = ['text/uri-list'];
+
+	async provideDocumentDropEdits(
+		document: vscode.TextDocument,
+		position: vscode.Position,
+		dataTransfer: vscode.DataTransfer,
+		token: vscode.CancellationToken):
+		Promise<vscode.DocumentDropEdit | undefined> {
+
+		const enabled = vscode.workspace.getConfiguration('css', document).get('format.formatDroppedFiles', true);
+		if (!enabled) {
+			return;
+		}
+
+		if (token.isCancellationRequested) {
```

Comment:
[nit] You can remove this. CancellationTokens only need to be checked after an async call (any `await`)